### ISSUE TITLE
Rank with function syntax does not accept descending property

### ISSUE THREAD

**Issue by olahjendev** [knex/knex#4646]
# Environment

Knex version: 0.95.9
Database + version: pg: 8.7.1
OS: Mac

# Bug

I'm trying to use the rank() call in the way demonstrated below, but it does not accept 'desc'.

```
knex('users').select('*').rank('alias_name', function() {
  this.orderBy('email', 'desc').partitionBy('firstName')
})
```

by running the above code, I get an ascending order for the ranks, therefore it looks like orderyBy() ignores 'desc' as a second parameter when ran inside rank()'s callback. I solved the issue by using the raw form, but I feel like the above way should also support this:

`knex('users').select('*').rank('alias_name', knex.raw('order by ?? desc', ['email']))
`

**Comment by elhigu** [comment]
First query generates: 

```
select *, rank() over (order by \"email\" desc) as alias_name from \"users\"
```

 raw generates:

**Comment by mercmobily** [comment]
Could not reproduce a bug in compilation; the `'desc'` direction is preserved in the generated SQL across all supported dialects (PostgreSQL, MySQL 8/2, OracleDB, MSSQL, SQLite3).

The analytic compiler renders order directions when provided. See `lib/query/querycompiler.js:1169–1181`, where each order entry contributes either `column` or `column + ' ' + order`. The `Analytic` builder captures the order and partitions as you call `analytic.orderBy(...)`. In our local check, your example compiled to (PostgreSQL):

```sql
select *, rank() over (partition by "firstName" order by "email" desc) as alias_name from "users"
```

If you have a case where `'desc'` is ignored, it might be tied to the specific driver or a preceding `orderBy` affecting row presentation rather than the window frame. A self‑contained script would help us drill in.

Important notes for usage
- The `orderBy('...', 'desc')` inside the analytic callback only affects the `OVER (...)` clause; it does not sort the final result set. If you expect rows returned in rank order, also add an outer `orderBy('alias_name', 'desc')` on the query.
- If you omit `partitionBy(...)` in the callback, your raw variant will naturally compile without a `PARTITION BY` clause; that explains output differences between the two examples in the original thread.

Quick verification (current code base)
```js
const db = require('./')({ client: 'pg' });
db('users')
  .select('*')
  .rank('alias_name', (w) => w.orderBy('email', 'desc').partitionBy('firstName'))
  .toSQL().sql;
// select *, rank() over (partition by "firstName" order by "email" desc) as alias_name from "users"
```

As for the question about rank(), I have this generated writeup -- I actually had it done a while ago for another project, since I needed it. This might not be correct:

```
  What rank() does (SQL semantics)

  - Window function: evaluated per row over a window defined by OVER (...).
  - Partitioning: PARTITION BY splits rows into independent groups; ranking
    restarts from 1 in each group. If omitted, all rows are a single partition.
  - Ordering (required for meaningful ranking): ORDER BY inside OVER determines
    the rank order. Rows that tie on the ordering keys receive the same rank.
  - Gaps: If there are ties, RANK produces gaps. Example: values [10, 10, 9]
    produce ranks [1, 1, 3]. For no gaps, use DENSE_RANK. For unique sequence
    regardless of ties, use ROW_NUMBER.

  Important usage notes

  - ORDER BY is inside the window: rank() over (order by …) governs ranking
    only; it does not sort the final result set. If you want output sorted by
    rank, add an outer orderBy('rank_alias', 'asc'|'desc') to the query.
  - Without ORDER BY: behaviour is not useful (some engines allow it and return
    1 for every row; others may require ORDER BY). Always supply ORDER BY for
    predictable ranking.
  - NULL ordering is dialect‑specific for the window ORDER BY:
      - PostgreSQL: ASC puts NULLs last by default; use NULLS FIRST|LAST to
        override.
      - Oracle: ASC → NULLS LAST; DESC → NULLS FIRST by default (can override).
      - SQL Server: ASC → NULLs first; DESC → NULLs last (no NULLS FIRST|LAST).
      - MySQL: ASC treats NULL as lowest (first); DESC puts NULLs last.
        If null positioning matters, normalize your data or coalesce/CASE in the
        ORDER BY.

  Knex API (how to specify rank correctly)
  Three equivalent ways to define the window:

  1. Callback builder (recommended for clarity)

  knex('users')
    .select('*')
    .rank('user_rank', (w) => w.orderBy('score',
  'desc').partitionBy('group_id'));

  2. Raw window clause

  knex('users')
    .select('*')
    .rank('user_rank', knex.raw('order by ?? desc', ['score']));

  3. Object/array form

  knex('users')
    .select('*')
    .rank('user_rank', [{ column: 'score', order: 'desc' }], { column:
  'group_id' });

  Outer sort if you want rows in rank order

  knex('users')
    .select('*')
    .rank('user_rank', (w) => w.orderBy('score', 'desc'))
    .orderBy('user_rank', 'desc'); // this sorts result rows by the computed
  rank

  Ties vs. other rankers (quick reference)

  - RANK: ties share rank; gaps appear.
  - DENSE_RANK: ties share rank; no gaps.
  - ROW_NUMBER: no ties; every row gets a unique sequence number.

  Example of ties (score: [100, 100, 90])

  - RANK → [1, 1, 3]
  - DENSE_RANK → [1, 1, 2]
  - ROW_NUMBER → [1, 2, 3] (but the order depends on the window ORDER BY)

  Edge cases and advice

  - Multi‑column order: pass more than one orderBy to break ties
    deterministically if needed (e.g., .orderBy('score', 'desc').orderBy('id',
    'asc')).
  - Nulls: if ranking must push NULLs last for all dialects, consider explicit
    COALESCE in the orderBy or CASE expressions in a raw clause to normalize
    cross‑dialect behaviour.
  - Final presentation: always add an outer orderBy on the alias if your API
    needs the results physically sorted by rank; the window ORDER BY does not
    guarantee the returned row order.

  Bottom line

  - rank() works the same conceptually across engines: partition first
    (optional), order inside OVER (required for meaningful ranks), expect gaps
    on ties, and add an outer orderBy to sort results by the computed rank. Use
    DENSE_RANK or ROW_NUMBER if you want different tie behaviour.
```
